app/controllers/houseConsignment/index/items/document/TypeController.scala

```diff
@@ -0,0 +1,114 @@
+  implicit val sessionRepository: SessionRepository,
+  navigator: Navigator,
+  actions: Actions,
+  getMandatoryPage: SpecificDataRequiredActionProvider,
+  formProvider: SelectableFormProvider,
+  service: DocumentsService,
+  val controllerComponents: MessagesControllerComponents,
+  view: TypeView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val prefix: String = "houseConsignment.index.items.document.type"
+
+  def onPageLoad(arrivalId: ArrivalId, mode: Mode, houseConsignmentIndex: Index, itemIndex: Index, documentIndex: Index): Action[AnyContent] = actions
+    .requireData(arrivalId)
+    .async {
+      implicit request =>
+        getDocumentList(request.userAnswers, service, houseConsignmentIndex, itemIndex, documentIndex).map {
+          documentList =>
+            val form = formProvider(mode, prefix, documentList, houseConsignmentIndex.display, itemIndex.display)
+            val preparedForm = request.userAnswers.get(TypePage(houseConsignmentIndex, itemIndex, documentIndex)) match {
+              case None        => form
+              case Some(value) => form.fill(value)
+            }
+
+            Ok(view(preparedForm, request.userAnswers.mrn, arrivalId, mode, documentList.values, houseConsignmentIndex, itemIndex, documentIndex))
+        }
+    }
+
+  def onSubmit(arrivalId: ArrivalId, mode: Mode, houseConsignmentIndex: Index, itemIndex: Index, documentIndex: Index): Action[AnyContent] = actions
+    .requireData(arrivalId)
+    .async {
+      implicit request =>
+        getDocumentList(request.userAnswers, service, houseConsignmentIndex, itemIndex, documentIndex).flatMap {
+          documentList =>
+            val form = formProvider(mode, prefix, documentList, houseConsignmentIndex.display, itemIndex.display)
+            form
+              .bindFromRequest()
+              .fold(
+                formWithErrors =>
+                  Future.successful(
+                    BadRequest(
+                      view(formWithErrors, request.userAnswers.mrn, arrivalId, mode, documentList.values, houseConsignmentIndex, itemIndex, documentIndex)
+                    )
+                  ),
+                value => redirect(mode, value, houseConsignmentIndex, itemIndex, documentIndex)
+              )
+        }
+    }
+
+  private def getDocumentList(userAnswers: UserAnswers, service: DocumentsService, houseConsignmentIndex: Index, itemIndex: Index, documentIndex: Index)(
+    implicit hc: HeaderCarrier
+  ): Future[SelectableList[DocumentType]] =
+    userAnswers.get(TypePage(houseConsignmentIndex, itemIndex, documentIndex)) match {
+      case None => service.getDocuments()
+      case Some(value) =>
+        value.`type` match {
+          case DocType.Transport => service.getTransportDocuments()
+          case DocType.Support   => service.getSupportingDocuments()
+          case _                 => service.getDocuments()
+        }
+    }
```

Comment:
Could this maybe live in the documents service?

Reply:
We also implement this logic in the document type page (Consignment level), I was thinking the service will become bloated if we move both *getDocumentList* methods in there. What do you think?

Reply:
I think I'm ok with that. It's easier to test this logic at the service level than the controller level